services/venice-controller/src/main/java/com/linkedin/venice/controller/kafka/TopicCleanupService.java

```diff
@@ -253,14 +253,24 @@ void cleanupVeniceTopics() {
           }
           boolean canDelete = true;
           for (Map.Entry<String, Map<String, Integer>> mapEntry: multiDataCenterStoreToVersionTopicCount.entrySet()) {
-            if (mapEntry.getValue().containsKey(topic.getStoreName())) {
-              canDelete = false;
-              LOGGER.info(
-                  "Topic deletion for topic: {} is delayed due to {} version topics found in datacenter {}",
-                  topic.getName(),
-                  mapEntry.getValue().get(topic.getStoreName()),
-                  mapEntry.getKey());
-              break;
+            String storeName = topic.getStoreName();
```

Comment:
This will only check local region's state. We need to check hybrid versions in other colos as well. See if you can leverage https://github.com/linkedin/venice/blob/14d05d439b404df2a6d7957f229be941899f7cb3/services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java#L3526